activation/handler.go

```diff
@@ -647,38 +637,49 @@ func (h *Handler) processATX(
 	return proof, err
 }
 
-// FetchReferences fetches referenced ATXs from peers if they are not found in db.
-func (h *Handler) FetchReferences(ctx context.Context, atx *types.ActivationTx) error {
-	if err := h.fetcher.GetPoetProof(ctx, atx.GetPoetProofRef()); err != nil {
-		return fmt.Errorf("atx (%s) missing poet proof (%s): %w",
-			atx.ShortString(), atx.GetPoetProofRef().ShortString(), err,
-		)
-	}
-
-	atxIDs := make(map[types.ATXID]struct{}, 3)
-	if atx.PositioningATX != types.EmptyATXID && atx.PositioningATX != h.goldenATXID {
-		atxIDs[atx.PositioningATX] = struct{}{}
+// registerHashes registers that the given peer should be asked for
+// the hashes of the poet proof and ATXs.
+func (h *Handler) registerHashes(peer p2p.Peer, poetRef types.Hash32, atxIDs []types.ATXID) {
```

Comment:
Have we already decided if an ATX will be able to contain multiple `poetRef`s?

Reply:
I think we did not. But if we decide so, this function can be adjusted to take a slice of poet-proof refs.